packages/gasket-plugin-intl/README.md

```diff
@@ -60,104 +60,242 @@ module.exports = {
+### intlLocale
 
-When determining what assets to precache, This plugin defaults to reading the
-first language provided in the `accept-language` header. However, you can
-override this behavior by adding an `intlLanguage` hook. The `intlLanguage` hook
+By default, the plugin will determine the locale from the first entry in the
+`accept-language` header of a request. However, you can override this behavior
+by implementing an `intlLocale` hook in an app or plugin. The `intlLocale` hook
 takes the following parameters:
 
-- `gasket` - (object) The Gasket config
-- `language` - (string) Default language specified by Gasket Intl
-- `req` - (string) The request
+- `gasket` - (object) Gasket session config
+- `locale` - (string) Default locale specified by Gasket Intl
+- `req` - (object) Request object
+- `req` - (object) Response object
 
-It should then return a string indicating the user's language, or null if this
-language cannot be found. This plugin will populate `intl.language` in the
-react state with this value, and use it for future language operations. If null
-is returned, Gasket will use `en-US`.
+It should then return a string indicating the user's locale. If no value is
+returned, Gasket will use `en-US`. Note that this is only available for Gasket
+apps with a server element, not for static sites.
 
 #### Example usage
 
 ```js
 module.exports = {
   hooks: {
-    intlLanguage: async function intlLanguageHook(gasket, language, req) {
+    intlLocale: async function intlLocaleHook(gasket, locale, req, res) {
       const { env } = gasket.config;
       // Always use en-US in dev for some reason....
       if(env === 'dev') return 'en-US';
       // This example could be handled via languageMap, but...
-      if(language.includes('fr')) {
+      if(locale.includes('fr')) {
         return 'fr-FR';
       }
-      // If no special cases apply, use the default language provided by Gasket.
-      return language;
+      // If no special cases apply, use the default locale provided by Gasket.
+      return locale;
     }
   }
 }
 ```
 
+## Module Files
 
-## How it works
-
-When you build your app, the plugin looks for `locale` directories in the app
-root and in node modules. Each file is then read and output to the build or
-outputDir with a hash added to the filename. This hash is based on the contents
-of the locale file, so if there are any changes, the file will get a new hash to
-ensure it is unique.
+There are several strategies for managing locale files for an app. By default,
+the plugin assumes they will be static files committed to the app's under a
+`public/locales` directory.
 
-Locale files are served with a long cache expiration time. This allows browsers
-to cache the locale files, and only versions with hash name changes will be
-newly downloaded.
+Another practice is to locale files under different npm modules. By enabling the
+`intl.modules` option in the `gasket.config.js`, when the app builds, the plugin
```

Comment:
Is the `intl.modules` option mandatory for the files to be picked up on build?

Reply:
Yes, it's an opt-in feature now.